pkg/resourceinterpreter/default/thirdparty/thirdparty_test.go

```diff
@@ -52,101 +58,155 @@ func checkScript(script string) error {
-	if err != nil {
-		t.Fatal(err)
-	}
-	obj := make(map[string]interface{})
-	err = json.Unmarshal(jsonData, &obj)
-	if err != nil {
-		t.Fatal(err)
-	}
-	return &unstructured.Unstructured{Object: obj}
-}
-
-func getAggregatedStatusItems(t *testing.T, path string) []workv1alpha2.AggregatedStatusItem {
-	if path == "" {
-		return nil
-	}
-	data, err := os.ReadFile(path)
-	if err != nil {
-		t.Fatal(err)
-	}
-	var statusItems []workv1alpha2.AggregatedStatusItem
-	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(data), 4096)
-	for {
-		statusItem := &workv1alpha2.AggregatedStatusItem{}
-		err = decoder.Decode(statusItem)
-		if err != nil {
-			break
-		}
-		statusItems = append(statusItems, *statusItem)
-	}
-	if err != io.EOF {
-		t.Fatal(err)
-	}
-
-	return statusItems
-}
-
 type TestStructure struct {
 	Tests []IndividualTest `yaml:"tests"`
 }
 
 type IndividualTest struct {
-	DesiredInputPath  string `yaml:"desiredInputPath,omitempty"`
-	ObservedInputPath string `yaml:"observedInputPath,omitempty"`
-	StatusInputPath   string `yaml:"statusInputPath,omitempty"`
-	DesiredReplica    int64  `yaml:"desiredReplicas,omitempty"`
-	Operation         string `yaml:"operation"`
+	Name          string                              `yaml:"name"`                    // the name of individual test
+	Description   string                              `yaml:"description,omitempty"`   // the description of individual test
+	DesiredObj    *unstructured.Unstructured          `yaml:"desiredObj,omitempty"`    // the desired object
+	ObservedObj   *unstructured.Unstructured          `yaml:"observedObj,omitempty"`   // the observed object
+	StatusItems   []workv1alpha2.AggregatedStatusItem `yaml:"statusItems,omitempty"`   // the status items of aggregated status
+	InputReplicas int64                               `yaml:"inputReplicas,omitempty"` // the input replicas for revise operation
+	Operation     string                              `yaml:"operation"`               // the operation of resource interpreter
+	Filepath      string                              `yaml:"filepath,omitempty"`      // the file path of current test case, used for logging
+	// TODO(@zhzhuang-zju): When we have a complete set of test cases, change Output to required field.
+	Output map[string]interface{} `yaml:"output,omitempty"` // the expected output results
 }
 
 func checkInterpretationRule(t *testing.T, path string, configs []*configv1alpha1.ResourceInterpreterCustomization) {
 	ipt := declarative.NewConfigurableInterpreter(nil)
 	ipt.LoadConfig(configs)
 
 	dir := filepath.Dir(path)
-	yamlBytes, err := os.ReadFile(dir + string(os.PathSeparator) + "customizations_tests.yaml")
-	if err != nil {
-		t.Fatal(err)
+	testDataDir := filepath.Join(dir, "testdata")
+
+	var err error
+	for _, customization := range configs {
+		for _, input := range getAllTestCases(t, testDataDir).Tests {
+			t.Run(fmt.Sprintf("[%s]:%s", input.Operation, input.Name), func(t *testing.T) {
```

Comment:
It is recommended to add a `customization.name` in the `fmt.Sprintf` info, which allows for quick identification of the test script in case of errors.